lib/class-wp-theme-config-resolver.php

```diff
@@ -289,23 +260,20 @@ public static function get_core_data() {
 	 *
 	 * @return WP_Theme_JSON Entity that holds theme data.
 	 */
-	public static function get_theme_data( $theme_support_data = array() ) {
-		if ( null === self::$theme ) {
-			$theme_json_data = self::read_json_file( self::get_file_path_from_theme( 'experimental-theme.json' ) );
-			$theme_json_data = self::translate( $theme_json_data, wp_get_theme()->get( 'TextDomain' ) );
-			self::$theme     = new WP_Theme_JSON( $theme_json_data );
-		}
-
+	private static function get_theme_data( $theme_support_data = array() ) {
```

Comment:
I use `get_theme_data` method in the test I added in #29828 (still waiting for review), but I'm sure it can be changed to use more higher level API.